flutter/flutter
# [windows] implement regular windows and supporting method channels

This pull request only makes changes to the windows embedder API and wires a flag through the framework. This does not include any user-facing API changes.

Design doc: https://docs.google.com/document/d/1eQG-IS7r4_S9_h50MY_hSGwUVtgSTRSLzu7MLzPNd2Y/edit?tab=t.0

## To test
1. Check out [this branch in canonical/flutter](https://github.com/canonical/flutter/pull/36) (this includes changes to the framework which are not included here on purpose for reviewer's sake!)
2. Build the engine
3. Then run the multi window reference application:
   ```
   cd examples/multi_window_ref_app/
   flutter run --debug --local-engine-src-path <PATH_TO_ENGINE_SRC> --local-engine host_debug_unopt --local-engine-host host_debug_unopt .\lib\main.dart --enable-multi-window
   ```
Don't forget the `--enable-multi-window` flag or it will not work

## What's new?

### Framework
- Added a new `--enable-multi-window` flag to turn on multi window features

### Engine
- Implemented regular windows in the windows embedder

![image](https://github.com/user-attachments/assets/dcfe0fd9-d631-4887-a92f-c6202e9a4fea)


## Pre-launch Checklist
